installer/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SaaSBoostInstall.java

```diff
@@ -380,23 +380,29 @@ private void installMetrics(String sbStackName) throws IOException {
                 .build();
         buildAndCopyLambdas("resources" + File.separator + "custom-resources", list);
 
-        //create db password if metrics installed
-        if (null == dbPassword || dbPassword.isEmpty()) {
-            UUID uniqueId = UUID.randomUUID();
-            String[] parts = uniqueId.toString().split("-");  //UUID 29219402-d9e2-4727-afec-2cd61f54fa8f
-            dbPassword = "Pass12" + parts[0];
-            //outputMessage("The database password for Redshift Metrics database is: " + this.dbPassword);
-        }
-
-        //create secure SSM parameter for password
+        //check if REDSHIFT_MASTER_PASSWORD param is already present so we don't create a new version
         ssm = SsmClient.builder()
                 .credentialsProvider(DefaultCredentialsProvider.builder().build())
                 .region(AWS_REGION).build();
-        // /saas-boost/${Environment}/METRICS_ANALYTICS_DEPLOYED to true
-        LOGGER.info("Update SSM param REDSHIFT_MASTER_PASSWORD with dbPassword");
-        //*TODO: check if parameter exists and do not update if it exists.
-        software.amazon.awssdk.services.ssm.model.Parameter passwordParam = putParameter(toParameterStore("REDSHIFT_MASTER_PASSWORD", dbPassword ,true));
-        dbPasswordSSMParameter = passwordParam.name();
+        try {
+            dbPasswordSSMParameter = "/saas-boost/" + this.envName + "/REDSHIFT_MASTER_PASSWORD";
+            GetParameterResponse parameterResponse = ssm.getParameter(GetParameterRequest.builder()
+                    .name(dbPasswordSSMParameter)
+                    .withDecryption(true)
+                    .build());
+            dbPassword = parameterResponse.parameter().value();
+            LOGGER.info("Found existing SSM Parameter: " + dbPasswordSSMParameter);
+        } catch (Exception e) {
+            //create db password param if not existing
+            LOGGER.info("SSM Parameter " + dbPasswordSSMParameter + " not found, add it.");
+            if (null == dbPassword || dbPassword.isEmpty()) {
+                UUID uniqueId = UUID.randomUUID();
+                String[] parts = uniqueId.toString().split("-");  //UUID 29219402-d9e2-4727-afec-2cd61f54fa8f
+                dbPassword = "Pass12" + parts[0];
```

Comment:
It looks like this "Pass12" was added to guarantee meeting [Redshift's password requirements](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_USER.html#r_CREATE_USER-parameters):

> * It must be 8 to 64 characters in length.
> * It must contain at least one uppercase letter, one lowercase letter, and one number.
> * It can use any ASCII characters with ASCII codes 33–126, except ' (single quotation mark), " (double quotation mark), \, /, or @.
Can you add a comment here to explain why a constant value is prepended to all passwords? Also, since UUID is always 36 characters, why not use the whole thing? This would increase the complexity of passwords, making them less vulnerable to attacks.